### ISSUE TITLE
Javascript redirects inside callbacks fail after cy.visit call with non-empy path

### ISSUE THREAD

**Issue by danieljabailey** [cypress-io/cypress#5717]
### Current behavior:

Suppose I have this in a spec.js:

```js
cy.visit('http://localhost:3000/subdir/index.html')
```

And something like this on the index.html page:

```js
setTimeout(()=>{
  window.location.href = '?done';
},1);
```

When running the test, the js redirect request that should go to `http://localhost:3000/subdir/index.html?done` actually goes to `http://localhost:3000/__/?done`

### Desired behavior:

Browser should redirect to the correct page

### Steps to reproduce: (app code and test code)

I have forked cypress-test-tiny and added a failing test case: https://github.com/danieljabailey/cypress-test-tiny

See the readme on that fork. The commit that adds the test case is summarised here: https://github.com/danieljabailey/cypress-test-tiny/commit/3ec55895b5556927d43637ed932502872b77812d

### Versions

Cypress: 3.6.1
Chrome: 78.0.3904.97 (Official Build) (64-bit)
Ubuntu: 18.04.3

**Comment by danieljabailey** [comment]
Another thing to note...

Without setTimeout wrapping the redirect: (button click does redirect immediately) this doesn't fail, it works as expected

**Comment by danieljabailey** [comment]
Probably the same issue as this: #3975

**Comment by jennifer-shehane** [comment]
Yes this is a duplicate of https://github.com/cypress-io/cypress/issues/3975 Closing as duplicate.

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
